### DISCUSSION TITLE
Only results if title STARTS with search term

### DISCUSSION THREAD

**Discussion by ltguillaume** [photoprism/photoprism#3906] [Q&A]
#### 1. What is not working as documented?

Searching for photos that contain a name _only_ seems to show results where the title of that photo _begins_ with that name, while hiding photos that have this name/the search term.

#### 2. How can we reproduce it?

1. Find a photo that has e.g. the title "Bob, Alice & Charlie"
2. Click on Search
3. Type in `Bob`
4. See the photo of "Bob, Alice & Charlie"
5. Type in `Alice`
6. Do **NOT** see the photo of "Bob, Alice & Charlie" listed
7. Type in `Charlie`
8. Do **NOT** see the photo of "Bob, Alice & Charlie" listed
9. Type in `*Charlie*`
10. Do **NOT** see the photo of "Bob, Alice & Charlie" listed

#### 3. What behavior do you expect?

If a search term is present _anywhere_ in at least title or description, the respective photo should be listed in the result.

#### 4. What could be the cause of your problem?

The search only checks if the title/description field begins with the search term.

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: Build 231021-9abea5b55 AMD64 Docker

(b) Database Type & Version: MariaDB

(c) Operating System Types & Versions: Linux

(d) Browser Types & Versions: all

(e) Ad Blockers, Browser Plugins, and/or Firewall Software? No

**Comment by graciousgrey** [comment]
Unfortunately I cannot reproduce this. Searching for each name showed the photo. Do you observe this for all photos with similar titles?

**Reply by ltguillaume** [reply]
You're right, it's not for all _names_. An example of a name that shows this issue is "May". I'm mentioning this name, because it's not just a name, but also a month, so perhaps it's also a keyword for the search engine or something?

What's the logic behind the search procedure?
1. Does it look in titles and description? Or all fields?
2. Does it search auto-generated titles?
3. Does it search the persons named via face recognition?

Another thing I see is that "literal strings", between quotes, don't seem to work, e.g. `"Bob, Alice & Charlie"`. That would really help, too, e.g. for "Firstname Lastname" searches.
